chcemvediet/apps/inforequests/admin.py

```diff
@@ -315,3 +324,92 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'email')
         return queryset
+
+    def get_actions(self, request):
+        actions = super(ActionAdmin, self).get_actions(request)
+        if u'delete_selected' in actions:
+            del actions[u'delete_selected']
+        return actions
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None or (obj.type not in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST] and
+                len(obj.branch.actions) > 1):
+            return True
+        return False
+
+    def delete_view(self, request, object_id, extra_context=None):
+        "The 'delete' admin view for this model."
+        opts = self.model._meta
+        app_label = opts.app_label
+
+        obj = self.get_object(request, unquote(object_id))
+
+        if not self.has_delete_permission(request, obj):
+            raise PermissionDenied
+
+        if obj is None:
+            raise Http404(
+                _('%(name)s object with primary key %(key)r does not exist.') %
+                {'name': force_text(opts.verbose_name), 'key': escape(object_id)}
+            )
+
+        using = router.db_for_write(self.model)
+
+        # Populate deleted_objects, a data structure of all related objects that
+        # will also be deleted.
+        (deleted_objects, perms_needed, protected) = get_deleted_objects(
+            [obj], opts, request.user, self.admin_site, using)
+        insert_inforequestemail(deleted_objects)
+
+        if request.POST:  # The user has already confirmed the deletion.
+            if perms_needed:
+                raise PermissionDenied
+            obj_display = force_text(obj)
+            self.log_deletion(request, obj, obj_display)
+            self.delete_model(request, obj)
+
+            return self.response_delete(request, obj_display)
+
+        object_name = force_text(opts.verbose_name)
+
+        if perms_needed or protected:
+            title = _("Cannot delete %(name)s") % {"name": object_name}
+        else:
+            title = _("Are you sure?")
+
+        context = dict(
+            self.admin_site.each_context(),
+            title=title,
+            object_name=object_name,
+            object=obj,
+            deleted_objects=deleted_objects,
+            perms_lacking=perms_needed,
+            protected=protected,
+            opts=opts,
+            app_label=app_label,
+            preserved_filters=self.get_preserved_filters(request),
+        )
+        context.update(extra_context or {})
+        return self.render_delete_form(request, context)
+
+    def delete_model(self, request, obj):
+        # delete action.email.inforequestemail_set.all()
+        super(ActionAdmin, self).delete_model(request, obj)
+
+def insert_inforequestemail(deleted_objects):
+    for i in range(len(deleted_objects)):
+        obj = deleted_objects[i]
+        m = re.match(r'Action: <a href=\".*>\[(\d+)\]', unicode(obj))
```

Comment:
Toto je velmi krehke. Zoznam zmazanych action nezistuj takto pracne parsovanim stringov, ktore mozu kedykolvek v buducnosti zmenit strukturu. Namiesto toho tiez pouzi `NestedObjects`, alebo priamo `Collector`, na vratenie priamo samotnych instancii.

Rovnako potom pri mazani v `delete_model` si najprv zisti zoznam action, ktora sa pojdu zmazat a na zaklade toho zoznamu si zisti, ktore `InforequestEmail` treba zmazat alebo updatnut.

Reply:
@martinmacko47 Ano, tuto cast (aj ostatne) som nechcel nechat tak ako to je, sluzilo to len pre moje ucely aby som si vyrobil nejaky hruby navrh ako by to mohlo vyzerat a fungovat. Spravil som ten jeden commit aby si videl ako to je, a chcel som hlavne napisat tie dve otazky, aby si videl aj konkretny kod k tym otazkam.

Reply:
:ok: